### DISCUSSION TITLE
Log Panel+: show persistent, time-synced logs, even during timeline navigation

### DISCUSSION THREAD

**Discussion by AlexTLeo** [foxglove/community#934] [Q&A]
At our company, we already have an internal tool for reading our robots' logs and debugging which is aimed at in-depth, expert debugging, and we are interested in deploying Foxglove as a first-level support tool for easier, simpler and more visual debugging (to be used by non robotics experts). 

We currently face issues with the log panel, which we have set to read the /rosout topic from a rosbag:
1.  The log panel is reading the log messages on-the-fly, which means we cannot see the fully available log at any given time, but would have to press play and wait for the messages to come in one by one.
2. All previously-received log messages on the panel get cleared if the user moves around the timeline.

A possible workaround that I have in mind is to first extract all the /rosout messages from the .bag and save the complete log to a .json, then read this .json from a custom Foxglove panel where it is formatted and displayed. This custom log panel should also be synchronised to the current playback time so that as you scroll, the bag itself moves in time as well to keep everything in time-sync.

We believe this feature could actually be beneficial to all to have in the official Foxglove release, rather than as our own personal hacky solution. A Log Panel+, if you will. We also wanted to know if this issue has been raised before (we couldn't find it specifically) and if a solution already exists that we were unaware of.

**Comment by rg-gravis** [comment]
+1 from us, my team has been wishing that the logs in the panel persisted when jumping around the timeline.

**Reply by AlexTLeo** [reply]
I'm afraid Foxglove is now a proprietary product and no longer open source!

**Reply by defunctzombie** [reply]
It's still a good feature request and we appreciate the product feedback.
